In [15]:
import pandas as pd

'../data/data2.csv'

In [22]:
df = pd.read_csv('../data/data2.csv')

In [23]:
df.shape

(7688, 20)

In [24]:
df.head(3)

,년도,월,항공사(ICAO),항공사(IATA),항공사명,국제_국내,노선,국가,공항,도착_출발,정기_부정기,여객_화물,운항(편),유임승객(명),무임승객(명),환승객(명),직화물(kg),환적화물(kg),우편물(kg),수하물(kg)
0,2026년,01월,AAL,AA,아메리칸항공,국제,미주,미국,DFW,도착,정기,여객,21,4108,16,681,11848,0,1632,115880
1,2026년,01월,AAL,AA,아메리칸항공,국제,미주,미국,DFW,출발,정기,여객,21,3788,14,1862,173237,0,14917,133829
2,2026년,01월,AAL,AA,아메리칸항공,국제,미주,미국,LAS,도착,정기,여객,3,552,0,11,1398,0,0,10980


In [25]:
df.columns.to_list()

['년도',
 '월',
 '항공사(ICAO)',
 '항공사(IATA)',
 '항공사명',
 '국제_국내',
 '노선',
 '국가',
 '공항',
 '도착_출발',
 '정기_부정기',
 '여객_화물',
 '운항(편)',
 '유임승객(명)',
 '무임승객(명)',
 '환승객(명)',
 '직화물(kg)',
 '환적화물(kg)',
 '우편물(kg)',
 '수하물(kg)']

In [ ]:
df.info()

# 2026년 상반기 인천공항에서 화물 전용 항공편은 전체 운항에서 얼마나 차지하며, 환적화물 운송에는 얼마나 기여하는가?

- 전체 운항 구조 파악


In [51]:
total_flights = df['운항(편)'].sum()
total_flights

np.int64(214645)

In [52]:
cargo_flights = df[df['여객_화물'] == '화물']['운항(편)'].sum()

cargo_flights

np.int64(24828)

In [53]:
cargo_ratio = (cargo_flights/total_flights) * 100

print(f'전체 운항편 중 화물 운항편 비율: {cargo_ratio:.2f}%')

전체 운항편 중 화물 운항편 비율: 11.57%


- 2026년 1월부터 6월까지 전체 운항편 중 화물 전용 운항편은 11.57%이다.

> 이 11.57%의 화물 전용 운항편은 어떤 항공사로 구성되어 있을까?
 - 화물 전용편을 운항한 항공사 목록은?
  - 그 중 화물 운항편이 많은 항공사는 어디?

In [57]:
filtered  = df[df['여객_화물'] == '화물']

In [58]:
filtered['여객_화물'].unique()

<StringArray>
['화물']
Length: 1, dtype: str

In [62]:
cargo_by_airline = filtered.groupby('항공사명')['운항(편)'].sum().sort_values(ascending= False)

cargo_by_airline

항공사명
대한항공                     6845
에어제타                     5012
FedEX항공                  2136
아틀라스항공                   2088
유피에스항공                   1503
칼리타항공                    1263
장쑤징동화물항공사                 556
슌펑(순풍) 항공                 481
에어로로직                     365
롱하오 항공                    362
싱가폴항공                     304
에어홍콩                      304
루프트한자항공                   303
전일본항공                     300
중국우정항공                    294
일본항공                      284
중국센트럴항공                   279
원통화물항공                    258
폴라에어카고                    250
카고룩스 이탈리아                 188
MAS AIR                   186
카타르항공                     168
카고룩스항공                    150
에티오피아 항공                  146
캐세이퍼시픽항공                  140
실크웨이웨스트항공                 136
중국화물항공                    134
터키항공                      104
마틴에어카고                     74
투르크메니스탄 항공                 54
스카이 리스                     30
델타항공                       27
웨스턴글로벌항공                   26
몽골항공 

화물 전용편은 전체 운항의 11.57%에 불과하지만, 전체 환적화물 운송에서는 얼마나 기여하고 있을까?

- 화물 전용기의 환적화물 기여율
- 여객기와 화물기의 편당 환적화물량 비교
- 항공사별 화물 운항편 수 순위(자주 운항한), 환적화물량 순위(많이 운송한) 확인

In [65]:
cargo_transfer = filtered['환적화물(kg)'].sum()
total_transfer = df['환적화물(kg)'].sum()

In [68]:
transfer_ratio = ((cargo_transfer/total_transfer)* 100)

print(f'전체 환적화물 중 화물 전용기의 기여율: {transfer_ratio:.2f}%')

전체 환적화물 중 화물 전용기의 기여율: 73.42%


In [71]:
cargo_per_flight = cargo_transfer / cargo_flights
cargo_per_flight

np.float64(16248.360963428388)

In [73]:
print(f'화물기편당환적화물량: {cargo_per_flight:.2f}kg/편')

화물기편당환적화물량: 16248.36kg/편


> 대략 화물기 한 편당 16,248kg의 환적화물을 운송했다..